File: backend/anvaya/schemas/health.py
from typing import Literal

from pydantic import BaseModel, ConfigDict


class HealthData(BaseModel):
    model_config = ConfigDict(extra="forbid")

    status: Literal["ok"]
    service: Literal["anvaya-api"]
    environment: Literal["development", "testing", "production"]
    database: Literal["ok"]
    public_demo_enabled: bool = False
    ai_assist_enabled: bool = False
    voice_enabled: bool = False
    ai_service_status: Literal["ok", "degraded", "unavailable", "disabled"] = "disabled"
    ai_service_message: str = ""
    voice_service_status: Literal["ok", "degraded", "unavailable", "disabled"] = "disabled"
    voice_service_message: str = ""

File: frontend/src/api/health.ts
export type HealthEnvelope = {
  request_id: string
  data: {
    status: 'ok'
    service: 'anvaya-api'
    environment: 'development' | 'testing' | 'production'
    database: 'ok'
  }
  warnings: string[]
}

export async function fetchHealth(): Promise<HealthEnvelope> {
  const response = await fetch('/api/health', {
    headers: { Accept: 'application/json' },
  })
  if (!response.ok) {
    throw new Error('The ANVAYA API is unavailable.')
  }
  return response.json() as Promise<HealthEnvelope>
}
